### DISCUSSION TITLE
Possible actions

### DISCUSSION THREAD

**Discussion by Fafois** [OlivierZal/com.heatzy#73] [General]
Hi,

Among the possible actions through the Heatzy's API, there is the activation or deactivation of a module's programmation, is it planed to implement it in the possible actions of the heatzy's Homey App?

Thanks

**Comment by OlivierZal** [comment]
Hi @Fafois 

I will search into the API documentation but indeed it should be possible to create a button or setting saying: ignore/unignore programmation. 

However I didn't dig in this possibility because I found it was a bit against Homey philosophy which prefers the programmation to be done via Homey flows rather than an external tool – with possible conflicts since automation could be operated by 2 different tools.

**Reply by Fafois** [reply]
Thanks for your response.

I have personally implemented it through direct calls to the API (because there was no App until now ;) ).
To do it you have to make a POST call to _https://euapi.gizwits.com/app/control_ with this body :
 _{"attrs": {"timer_switch":1}}_ to switch programmation on
_{"attrs": {"timer_switch":0}}_ to switch programmation off

Best regards
François

**Comment by OlivierZal** [comment]
Hi @Fafois, a new code update is ready and takes into account the timer switch but also other actions available from the API (everything which is not related to timer configuration): https://github.com/OlivierZal/com.heatzy/commit/3c6bc22461040f84c4a9c0a405cbf846a775c02b

I'll deploy in the next days.

**Reply by Fafois** [reply]
Bonjour Olivier, merci beaucoup, c'est très gentil à vous.


Hi Olivier, thanks a lot, it's very nice.

**Reply by Fafois** [reply]
Hi Olivier.

Do you know when you will publish the new version please ?

Thank you

**Comment by OlivierZal** [comment]
Hi @Fafois, there was lack in Heatzy's documentation (because I also implement boost mode, etc.) and they answered me some days ago, I think I'll deploy by end of the week.

**Reply by Fafois** [reply]
Great, thanks a lot.

**Comment by OlivierZal** [comment]
@Fafois, "voilà": https://homey.app/fr-fr/app/com.heatzy/Heatzy/test/

**Reply by Fafois** [reply]
Super, merci :)

**Comment by Fafois** [comment]
Bonjour Olivier. J'ai testé la dernière version mais je ne trouve toujours pas la possibilité d'activer/désactiver le programme en cours.
Pour rappel pour activer le programme en cours sur leur API c'est : {"attrs": {"timer_switch":1}}
Pour le désactiver c'est : {"attrs": {"timer_switch":0}}

Si j'ai mal utilisé l'app n'hésite pas à me le dire.

Merci
